app/packages/looker-3d/src/utils.test.ts

```diff
@@ -0,0 +1,96 @@
+import { BufferAttribute, Quaternion, Vector3 } from "three";
+import { describe, expect, it } from "vitest";
+import { COLOR_POOL } from "./constants";
+import {
+  computeMinMaxForColorBufferAttribute,
+  computeMinMaxForScalarBufferAttribute,
+  deg2rad,
+  getColorFromPoolBasedOnHash,
+  getGridQuaternionFromUpVector,
+  toEulerFromDegreesArray,
+} from "./utils";
+
+describe("deg2rad", () => {
+  it("converts degrees to radians", () => {
+    expect(deg2rad(0)).toBe(0);
+    expect(deg2rad(180)).toBeCloseTo(Math.PI);
+    expect(deg2rad(90)).toBeCloseTo(Math.PI / 2);
+  });
+});
+
+describe("toEulerFromDegreesArray", () => {
+  it("converts array of degrees to radians", () => {
+    expect(toEulerFromDegreesArray([0, 90, 180])).toEqual([
+      0,
+      Math.PI / 2,
+      Math.PI,
+    ]);
+  });
+});
+
+describe("computeMinMaxForColorBufferAttribute", () => {
+  it("computes min and max for color attribute", () => {
+    const attr = new BufferAttribute(new Float32Array([1, 2, 3, 4, 5, 6]), 1);
+    expect(computeMinMaxForColorBufferAttribute(attr)).toEqual({
+      min: 0,
+      max: 6,
+    });
+  });
+  it("handles all negative values", () => {
+    const attr = new BufferAttribute(new Float32Array([-1, -2, -3]), 1);
+    expect(computeMinMaxForColorBufferAttribute(attr)).toEqual({
+      min: -3,
+      max: 0,
+    });
+  });
+});
+
+describe("computeMinMaxForScalarBufferAttribute", () => {
+  it("computes min and max for scalar attribute", () => {
+    const attr = new BufferAttribute(new Float32Array([1, 2, 3, 4, 5, 6]), 1);
+    expect(computeMinMaxForScalarBufferAttribute(attr)).toEqual({
+      min: 1,
+      max: 6,
+    });
+  });
+  it("handles negative values", () => {
+    const attr = new BufferAttribute(new Float32Array([-1, -2, -3, 0]), 1);
+    expect(computeMinMaxForScalarBufferAttribute(attr)).toEqual({
+      min: -3,
+      max: 0,
+    });
+  });
+  it("handles empty array", () => {
+    const attr = new BufferAttribute(new Float32Array([]), 1);
+    expect(computeMinMaxForScalarBufferAttribute(attr)).toEqual({
+      min: Infinity,
```

Comment:
nit:

This seems weird. I would expect the defaults to be in the actual attribute ranges supported by the shaders or null/undefined. Infinity here is confusing.